### PR TITLE
Cu 864ecru3t create settings page configurations

### PR DESCRIPTION
**Pull request by vinicius-prates**
##### Settings page at account, new feature that the user can change his email, role and area that he is working on

- [X] Why (Why did you make this pull request): Created new feature, settings page

- [X] Brief description of the changes: New feature that can change email, role and area of the own user.
- [X] Possible breakouts (Possible places where things could break): In the Input button, if someone made any modification at the component

### CODE REVIEW

File: apps/web-common/src/pages/account/settings.tsx [hunk 1]
```diff
@@ -1,7 +1,145 @@
...
+}: {
+  email: string;
+  cargo: string;
+  area: string;
+}) => {
+  const res = await fetch(authURL, {
+    method: "PATCH",
+    body: JSON.stringify({
+      email: email,
+      roleName: cargo,
```
Comment (vitormiura) [inline apps/web-common/src/pages/account/settings.tsx:22]: I think only admins should change people roles

**Review by vitormiura — CHANGES_REQUESTED** [review]
Overall that PR needs some changes, why common users would change their email, name and course?

File: apps/web-common/src/pages/account/settings.tsx (hunk 1 above)
Comment (viktormarinho) [reply apps/web-common/src/pages/account/settings.tsx:22]: ?

**Comment by viktormarinho** [comment]
> Overall that PR makes no sense, why common users would change their email, name and course? -1

Email and sector needs to be editable. Rolename should not.

**Comment by vitormiura** [comment]
> > Overall that PR makes no sense, why common users would change their email, name and course? -1
> 
> Email and sector needs to be editable. Rolename should not.

makes sense, i thought users will use bosch email

File: apps/web-common/src/pages/account/settings.tsx (hunk 1 above)
Comment (vitormiura) [reply apps/web-common/src/pages/account/settings.tsx:22]: jk with prates

**Comment by abehidek** [comment]
We should think better of this, since it has implications on the email notification system as well. @viktormarinho let's talk better after

File: apps/web-common/src/pages/account/settings.tsx (hunk 1 above)
Comment (vinicius-prates) [reply apps/web-common/src/pages/account/settings.tsx:22]: Indeed we should think better about this feature, what about just a Admin could change this infos?

**Comment by viktormarinho** [comment]
> > > Overall that PR makes no sense, why common users would change their email, name and course? -1
> > 
> > 
> > Email and sector needs to be editable. Rolename should not.
> 
> makes sense, i thought users will use bosch email

Users start with No email. after some time then they receive a microsoft account and can update their email field

**Comment by vitormiura** [comment]
> > Overall that PR makes no sense, why common users would change their email, name and course? -1
> 
> Email and sector needs to be editable. Rolename should not.

We need to think better about the apprentice email, cuz in the first 6 months they use the ets common email, and personal emails wouldnt be able to work

**Comment by viktormarinho** [comment]
> We should think better of this, since it has implications on the email notification system as well. @viktormarinho let's talk better after

Why would it affect the email system? when sending an email just get a list of users and send emails to each one of them using their current email. If there is no email simply send nothing

**Comment by abehidek** [comment]
> Why would it affect the email system? when sending an email just get a list of users and send emails to each one of them using their current email. If there is no email simply send nothing

We cannot send emails to gmail/outlook or any other outside email provider from Bosch because of proxy, we can only send emails to @bosch.com accounts

**Comment by abehidek** [comment]
Only focal points and people with the Bosch email would receive notifications then, which I don't find too good particularly

**Comment by abehidek** [comment]
One solution is to create a notification system inside Einer and use it across all applications (we could put it inside the Navbar or something idk)

**Comment by viktormarinho** [comment]
If you want to notify users that do not have an email yet we can do this, in a separate service or inside the auth service itself, making every user contain a list of notifications. I dont know which one of them i would prefer, but if we would make a new separate service for notifications, i think mailing logic should be there too. That way notifications can go to the user email as well in a easy way, they can disable/enable it, looks like a good alternative

**Comment by viktormarinho** [comment]
Consume the users notifications should be easy, too. Overall that is a service that makes sense.

**Comment by abehidek** [comment]
In theory, UI lib should only be used for generic composable components that represents basic primitives of the UI like the shadcn ui library.

However, I see that there are a lot of components that do some external querying/side-effect in the UI package, for the most part we should do this kinda stuff inside the components directory of the project rather than putting it in the UI lib, but for components that we share across applications, I think there isn't too much problem in doing so, we just need to not make this common across all components, and separate it from the pure/simple components.

a great example of component that is used in two applications but perform actions is `DeleteProjectModal`, instead of passing the mutation fn as a prop, we can just pass a event handler callback that will be called when the you click the delete button in the component, the implementation details on how you should do this is unknown to the component because you provide a fn to doing so.

tldr: usually components in ui should be generic, components with state should live inside the src/components or inside the page in the web app, but this is not a rule in general but we should generally avoid it to not make our components wacky

**Comment by abehidek** [comment]
We should talk about this later actually
